### PR TITLE
[JENKINS-58716] /project/scm/url may now point to a fork; relax check

### PR DESCRIPTION
**Pull request by jglick**
Required for https://github.com/jenkinsci/incrementals-tools/pull/16 to be useful. See deployment error on https://github.com/jenkinsci/workflow-step-api-plugin/pull/58 sample.

### CODE REVIEW

**Comment by jglick** [comment]
Is there anyone in INFRA who has the ability to merge PRs like this (and #22) and make sure they are deployed? @slide @timja et al.

File: incrementals-publisher/test/permissions-test.js [hunk 1]
```diff
@@ -25,13 +25,11 @@ describe('The Permissions helpers', function() {
...
         entries,
         perms,
-        folderMetadataParsed.owner,
-        folderMetadataParsed.repo,
         buildMetadataParsed.hash
       ),
       {
         name: 'Error',
-        message: 'ZIP error: Error: Wrong URL in /project/scm/url'
+        message: 'ZIP error: Error: Missing <url> section in <scm> of io/jenkins/tools/bom/bom/2.176.1-rc41.149af85f094d/bom-2.176.1-rc41.149af85f094d.pom'
```
Comment (slide) [inline incrementals-publisher/test/permissions-test.js:32]: Is this actually correct? Should it have that specific version of the bom in this message?

Comment (slide) [reply incrementals-publisher/test/permissions-test.js:32]: Oh, this is a test, I see. I was misreading.

**Comment by slide** [comment]
@rtyler Did you write this initially? Any input?

File: incrementals-publisher/test/permissions-test.js (hunk 1 above)
Comment (jglick) [reply incrementals-publisher/test/permissions-test.js:32]: The assertion here previously was verifying a condition that as of this PR is not considered incorrect, so I had to amend the test ZIP to delete just the `<url>` so as to still verify that _some_ checks are being done.

**Comment by jglick** [comment]
@rtyler wrote the original function, as a semi-port of a prototype I had written in Java, but it has had various editors since then.

**Pull request merged**